### ISSUE TITLE
`type-annotation-spacing` and `key-spacing`: conflict

### ISSUE THREAD

**Issue by Mqxx** [eslint-stylistic/eslint-stylistic#476]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

`key-spacing` should only check for key values not for types but `key-spacing` is also checking types. Types should only be checked by `type-annotation-spacing` and not `key-spacing`.

This is my current `eslint.config.js`:
```js
// eslint.config.js
import parserTs from '@typescript-eslint/parser'
import stylistic from '@stylistic/eslint-plugin'

export default [
  {
    files: ["**/*.ts"],
    plugins: {
      '@stylistic': stylistic
    },
    languageOptions: {
      parser: parserTs
    },
    rules: {
      '@stylistic/type-annotation-spacing': ['error', {
        "before": true,
        "after": true
      }],
      '@stylistic/key-spacing': ['error', {
        "beforeColon": false,
        "afterColon": true,
        "mode": "strict"
      }]
    }
  }
];
```

As an example, this is how I want my code to be formatted:
```ts
// deno-lint-ignore-file
let fooBar : string = "bar";
//        ^ ^ 1 space

function foo() : {e : string} {return {e: ''};}
//            ^ ^  ^ ^ 1 space          ^ No space

class Foo {
    name : string | undefined;
//      ^ ^ 1 space
}

type Fooo = () => {};
```

```
  4:19  error  Extra space after key 'e'     @stylistic/key-spacing
  7:5   error  Extra space after key 'name'  @stylistic/key-spacing

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

The rules described conflict.

### Reproduction

https://github.com/DenoPlayground/eslint-example

https://github.com/eslint/eslint/issues/18701

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by Morgul** [comment]
Just ran into this myself.

**Comment by Mqxx** [comment]
I think [this line](https://github.com/eslint-stylistic/eslint-stylistic/blob/dde2c6fb4a344b0dcf54679f2849b4cd25a741ee/packages/eslint-plugin-ts/rules/key-spacing/key-spacing.ts#L91) is the problem... The rule `key-spacing` should not check for `typeAnnotation` thats what the `type-annotation-spacing` rule is for.

**Comment by Mqxx** [comment]
Any update on this?

**Comment by VincentLKoch** [comment]
@antfu Sorry to ping you here, may you please take a look as the author seem to have identified the problem already ?
There is also the issue #491 that propose an alternative way to fix this
